app/models/appointment.rb

```diff
@@ -51,16 +51,26 @@ def self.between(start_date, end_date)
     where("scheduled_date BETWEEN ? and ?", start_date, end_date)
   end
 
-  def self.all_overdue
-    where(status: "scheduled")
-      .where(arel_table[:scheduled_date].lt(Date.current))
-      .where(arel_table[:remind_on].eq(nil).or(arel_table[:remind_on].lteq(Date.current)))
+  def self.passed_unvisited
+    # Scheduled or cancelled appointments whose scheduled date has passed.
+    where.not(appointments: {status: :visited})
+      .where("appointments.scheduled_date < ?", Date.current)
       .joins(:patient)
       .where.not(patients: {status: :dead})
   end
 
+  def self.last_year_unvisited
+    passed_unvisited.where("appointments.scheduled_date >= ?", 365.days.ago)
+  end
+
+  def self.all_overdue
+    passed_unvisited
+      .where(status: :scheduled)
+      .where(arel_table[:remind_on].eq(nil).or(arel_table[:remind_on].lteq(Date.current)))
+  end
+
   def self.overdue
-    all_overdue.where(arel_table[:scheduled_date].gteq(365.days.ago))
+    all_overdue.where("appointments.scheduled_date >= ?", 365.days.ago)
```

Comment:
The `overdue` scope is only used by the filters on the dashboard and can be done directly in `PatientSummaryQuery`. The names `overdue` and `all_overdue` are not clear. We think:
- Remove the `overdue` method
- Rename `all_overdue` to `overdue`
- No need for `last_year_unvisited` as well, do the filtering in `PatientSummaryQuery`

This is a larger change and can come in a follow up PR.

Reply:
I feel like getting these names right is a big piece of this work and future improvements. We have three scopes in diminishing order:
1. All overdue patients regardless of their "latest communication status" - to power the download
2. Overdue patients that need to be contacted right away - to power the dashboard and app lists
3. (2) that are not more than 365 days overdue - to power some filtering on the dashboard

I'm leaning towards calling (1) Overdue, and then setting up _independent_ scopes that can be composed for (2) and (3). Eg.

`Appointment.overdue`
`Appointment.overdue.require_reminder`
`Appointment.overdue.within_one_year_overdue.require_reminder`

The names are awkward, but the concept is that we don't introduce _nested_ scopes, we introduce _independent_ scopes that are chained together, which may lend itself to more intuitive names.

Reply:
That sounds good. The `overdue` methods have touch points all over that we wanted to avoid with this feature PR. 

> I'm leaning towards calling (1) Overdue
This would mean our "overdue" definition changes to be "scheduled appointment in the past" to "any appointment in the past" though.